### ISSUE TITLE
Add back py.typed

### ISSUE THREAD

**Issue by Mariatta** [googleapis/python-firestore#548]
Adding py.typed file is a breaking change and require a major version bump. We've removed `py.typed` file from the repo for now to unblock the release. We should add this back in in the future.
There should also be a corresponding `mypy` check in `noxfile.py` that tests the `google` and `samples` directory.

**Comment by meredithslota** [comment]
Closing in favor of consolidating the issues related to typing in #447
